OneSchedule/Controllers/UpdateController.cs

```diff
@@ -12,9 +12,9 @@ namespace OneSchedule.Controllers
     [ApiController]
     public class UpdateController : ControllerBase
     {
-        private readonly StrategyContext _context;
+        private readonly IStateContext _context;
 
-        public UpdateController(IService<UserDomain> service, IMapper mapper, StrategyContext context)
+        public UpdateController(IService<UserDomain> service, IMapper mapper, IStateContext context)
```

Comment:
Controller should use Strategy at first and based on request parameters you can choose the proper action